### ISSUE TITLE
Backwards compatibility: Handling of older JS runtimes, Browsers, ...

### ISSUE THREAD

**Issue by Querela** [giltayar/proposal-types-as-comments#76]
Ignoring the type hints according to the proposal only seems to target runtime performance (except for parsing), and adding metadata or introspection is not planned but backwards compatibility seems to be a major issue?

So, how would older browsers or JS runtimes in general handle this additional (still unsupported) syntax?
I'm not too familiar with the previous JavaScript history of feature additions, and would assume similar to arrow functions, new APIs (fetch) etc. But the landscape of browser types and versions varies a lot, so the new syntax would most likely fail on older systems.

Will there be a grace period (how long?) where we would use some transpiler to strip out the type hints until we can assume a majority of browsers supports them?
Or is this something not really concerning the proposal at all? And more for the TC39 committee to plan about?

**Comment by bakkot** [comment]
Any new syntax has the same problem, and it's always solved the same way: if you want to ship the new syntax to production but still support older browsers, you need to use a tool like babel to remove the new syntax before shipping. It's up to you to decide when enough people have upgraded that you can stop using babel.

**Comment by Querela** [comment]
Maybe to summarize: the new type syntax is ignored by browsers to allow unchanged execution of code (no type checking) but until browsers are able to parse and filter out the new syntax the code will not run?

**Comment by wparad** [comment]
> Maybe to summarize: the new type syntax is ignored by browsers to allow unchanged execution of code (no type checking) but until browsers are able to parse and filter out the new syntax the code will not run?

I think @bakkot pretty well summed up how it would be, but what you are sharing here doesn't seem to echo that, is there a reason you are trying to restate it?

**Comment by Querela** [comment]
> Any new syntax has the same problem, and it's always solved the same way: if you want to ship the new syntax to production but still support older browsers, you need to use a tool like babel to remove the new syntax before shipping. It's up to you to decide when enough people have upgraded that you can stop using babel.

Ok. So a non-issue for the proposal?

**Comment by orta** [comment]
Yes, it is a non-issue - this is how all TC39 proposals are handled 👍🏻

**Comment by Querela** [comment]
> > Maybe to summarize: the new type syntax is ignored by browsers to allow unchanged execution of code (no type checking) but until browsers are able to parse and filter out the new syntax the code will not run?
> 
> I think @bakkot pretty well summed up how it would be, but what you are sharing here doesn't seem to echo that, is there a reason you are trying to restate it?

Hmm. Well, if the browsers do support the syntax, then they could optionally have a "strict" mode to check for type invalid code?
I know that parsing and skipping some code is different compared to implementing a complete type checker.
